src/preact.d.ts

```diff
@@ -88,6 +88,8 @@ declare namespace JSX {
 	}
 
 	interface SVGAttributes {
+		class?:string | { [key:string]: boolean };
+		className?:string | { [key:string]: boolean };
```

Comment:
It currently supports accepting an object, and compiles a String from the keys with truthy values.  However, this is almost certainly being removed in Preact 8.